packages/rum-recorder/src/boot/recorder.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { createNewEvent, HttpRequest, isIE } from '@datadog/browser-core'
+import { LifeCycle, LifeCycleEventType } from '@datadog/browser-rum-core'
+
+import { setup, TestSetupBuilder } from '../../../rum-core/test/specHelper'
+
+import { startRecording } from './recorder'
+
+describe('startRecording', () => {
+  let setupBuilder: TestSetupBuilder
+  let sessionId: string | undefined
+  let waitRequests: (callback: (requests: ReadonlyArray<{ data: FormData; size: number }>) => void) => void
+
+  beforeEach(() => {
+    if (isIE()) {
+      pending('IE not supported')
+    }
+    sessionId = 'session-id'
+    setupBuilder = setup()
+      .withParentContexts({
+        findView() {
+          return {
+            session: {
+              id: sessionId,
+            },
+            view: {
+              id: 'view-id',
+              referrer: '',
+              url: 'http://example.org',
+            },
+          }
+        },
+      })
+      .beforeBuild(({ lifeCycle, applicationId, configuration, parentContexts, session }) => {
+        return startRecording(lifeCycle, applicationId, configuration, session, parentContexts)
+      })
+
+    const requestSendSpy = spyOn(HttpRequest.prototype, 'send')
+
+    waitRequests = (callback) => {
+      let isWaiting = false
+
+      requestSendSpy.and.callFake(() => {
+        if (isWaiting) {
+          return
+        }
+        isWaiting = true
+        setTimeout(() => {
+          callback(requestSendSpy.calls.allArgs().map(([data, size]) => ({ size, data: data as FormData })))
+        }, 300)
+      })
+    }
```

Comment:
```suggestion
    waitRequests = (callback) => {
      setTimeout(() => {
        callback(requestSendSpy.calls.allArgs().map(([data, size]) => ({ size, data: data as FormData })))
      }, 300)
    }
```
From local test, this also seems to work, do we need the extra logic?

Reply:
The issue I had with this is that sometimes 300ms is not enough so the test became flacky. This was handled nicely with the `throttle` function..

An alternative solution would be to specify the number of expected requests. `waitRequests` would wait for those requests, then wait 300ms to ensure no further request is being sent.